### PR TITLE
More correct distribution of tasks into blocks in role "k3s_server"

### PR DESCRIPTION
**Pull request by DmitriySafronov**
#### Changes ####
- Move control node tasks to separate block
- Change hardcoded "~/.kube/config.new" logic to use ansible variable
- Change kubeconfig editing tasks logic in control node tasks

#### Linked Issues ####

### CODE REVIEW

**Comment by dereknola** [comment]
Please open an issue or describe what problem you are attempting to address with this change.

File: roles/k3s_server/tasks/main.yml [hunk 1]
```diff
@@ -73,25 +73,11 @@
       ansible.builtin.pause:
         seconds: 10
 
-    - name: Add K3s autocomplete to user bashrc
-      ansible.builtin.lineinfile:
-        path: "~{{ ansible_user }}/.bashrc"
-        regexp: '\.\s+<\(k3s completion bash\)'
-        line: ". <(k3s completion bash)  # Added by k3s-ansible"
```
Comment (dereknola) [inline roles/k3s_server/tasks/main.yml:80]: This functionality is still useful, you should not remove it.

File: roles/k3s_server/tasks/main.yml [hunk 2]
```diff
@@ -100,24 +86,37 @@
...
+    - name: Copy kubeconfig to control node
+      when: kubectl_installed.rc == 0
+      ansible.builtin.fetch:
+        src: /etc/rancher/k3s/k3s.yaml
+        dest: "{{ kubeconfig }}"
+        flat: true
+
+    - name: Change server address in kubeconfig on control node
+      when: kubectl_installed.rc == 0
+      ansible.builtin.shell: KUBECONFIG={{ kubeconfig }} kubectl config set-cluster default --server=https://{{ api_endpoint | default(inventory_hostname) }}:{{ api_port }}
```
Comment (dereknola) [inline roles/k3s_server/tasks/main.yml:98]: `api_endpoint` is a required value, there should be any need to default to the inventory hostname.

File: roles/k3s_server/tasks/main.yml [hunk 3]
```diff
@@ -100,24 +86,37 @@
...
-      when: kubeconfig == "~/.kube/config.new" and kubectl_installed.rc == 0
+    - name: Merge with any existing kubeconfig on control node
+      when: kubeconfig != "~/.kube/config" and kubectl_installed.rc == 0
       ansible.builtin.shell: |
         TFILE=$(mktemp)
-        KUBECONFIG=~/.kube/config.new kubectl rename-context default k3s-ansible
-        KUBECONFIG=~/.kube/config.new kubectl config set-context k3s-ansible --user=k3s-ansible --cluster=k3s-ansible
-        KUBECONFIG=~/.kube/config.new:~/.kube/config kubectl config view --flatten > ${TFILE}
+        KUBECONFIG={{ kubeconfig }} kubectl config set-context k3s-ansible --user=k3s-ansible --cluster=k3s-ansible
+        KUBECONFIG={{ kubeconfig }} kubectl config use-context k3s-ansible
```
Comment (dereknola) [inline roles/k3s_server/tasks/main.yml:116]: I specifically avoided changing the users kubectl context when I added this functionality. I didn't want to modify the control plane more than necessary.

**Review by dereknola — CHANGES_REQUESTED** [review]

File: roles/k3s_server/tasks/main.yml (hunk 3 above)
Comment (DmitriySafronov) [reply roles/k3s_server/tasks/main.yml:116]: I'll remove "use-context". Will that be ok?

File: roles/k3s_server/tasks/main.yml (hunk 1 above)
Comment (DmitriySafronov) [reply roles/k3s_server/tasks/main.yml:80]: I'll move it into "control node" block

Comment (DmitriySafronov) [reply roles/k3s_server/tasks/main.yml:80]: Wait. This is just double for https://github.com/DmitriySafronov/k3s-ansible/blob/dd316c7caeadb3b4f9dfad9d072191362be5c578/roles/k3s_server/tasks/main.yml#L191 isn't it?

**Comment by DmitriySafronov** [comment]
> Please open an issue or describe what problem you are attempting to address with this change.

I want to have an option to disable all modifications to user environment, both local and remote:
- user_kubectl (alredy exists)
- control_kubectl (my edits)

File: roles/k3s_server/tasks/main.yml (hunk 1 above)
Comment (DmitriySafronov) [reply roles/k3s_server/tasks/main.yml:80]: This are the same tasks are they?
- [First #L76](https://github.com/k3s-io/k3s-ansible/blob/d1d7864337bfb527ae0e281276fe6d2f95ff14a2/roles/k3s_server/tasks/main.yml#L76)
- [Second #L192](https://github.com/k3s-io/k3s-ansible/blob/d1d7864337bfb527ae0e281276fe6d2f95ff14a2/roles/k3s_server/tasks/main.yml#L192)

Comment (DmitriySafronov) [reply roles/k3s_server/tasks/main.yml:80]: K3S server & control node now have own completion setup tasks

**Comment by DmitriySafronov** [comment]
Will rework PR

**Pull request closed without merging**